src/Inkluzitron/Modules/Vote/VotegagModule.cs

```diff
@@ -0,0 +1,211 @@
+
+        public VotegagModule(UsersService usersService, VoteSettings voteSettings,
+            BotSettings botSettings, IMemoryCache cache)
+        {
+            UsersService = usersService;
+            VoteSettings = voteSettings;
+            BotSettings = botSettings;
+            Cache = cache;
+        }
+
+        static private string FormatPeople(int number)
+        {
+            if (number == 1) return "člověk";
+            else if (number > 1 && number < 5) return "lidi";
+            return "lidí";
+        }
+
+        static public string GenerateVoteMessage(IUser target, int minutes, int minVotes, int votes, DateTime voteEnd)
+            => GenerateVoteMessage(target.Mention, minutes, minVotes, votes, voteEnd);
+
+        static public string GenerateVoteMessage(string targetMention, int minutes, int minVotes, int votes, DateTime voteEnd)
+        {
+            var text = new StringBuilder(
+                $"Hlasování o umlčení {targetMention} na dobu **{minutes} minut**.\n");
+
+            text.Append($"S návrhem musí souhlasit alespoň {minVotes} {FormatPeople(minVotes)}");
+            if (votes != 0) text.Append($" ({votes} {FormatPeople(votes)} {(votes > 1 && votes < 5 ? "jsou" : "je")} pro)");
+            text.Append($".\n*Hlasování bude ukončeno v {voteEnd:HH:mm}.*");
+
+            return text.ToString();
+        }
+
+        static public string CreateCacheKey(ulong userId)
+            => $"votegag_{userId}";
+
+        private async Task MuteUser(SocketGuildUser user, int minutes)
+        {
+            var muteRole = user.Guild.Roles.First(r => r.Id == VoteSettings.MuteRole);
+            await user.AddRoleAsync(muteRole);
+
+            var mutedUntil = DateTime.Now.AddMinutes(minutes);
+            await UsersService.SetMutedUntilAsync(user, mutedUntil);
+        }
+
+        static public async Task UnmuteUser(SocketGuildUser user, UsersService usersService, SocketRole muteRole)
+        {
+            await user.RemoveRoleAsync(muteRole);
+
+            await usersService.SetMutedUntilAsync(user, null);
+
+            try
+            {
+                await user.SendMessageAsync("Tvé umlčení právě skončilo.");
+            }
+            catch (HttpException)
+            {
+                // User has blocked DMs
+            }
+        }
+
+        [Command("votegag")]
+        [Summary("Zahájí hlasování o umlčení uživatele na stanovenou dobu v minutách (výchozí=30, min=5, max=240).")]
+        public async Task StartVoteAsync([Name("uživatel")] IUser user, [Name("minut")] int minutes = 30)
+        {
+            var minMinutes = 5;
+            var maxMinutes = 240;
+
+            if (minutes < minMinutes || minutes > maxMinutes)
+            {
+                await ReplyAsync("Doba umlčení musí být v rozsahu 5-240 minut.");
+                return;
+            }
+
+            if (Context.IsPrivate || Context.Guild.Id != BotSettings.HomeGuildId)
```

Comment:
Pokud chceš omezit použití příkazu jen na server, tak použij `RequireContextAttribute`. (Komentář se týká jen první části podmínky).